java-vtl-script/src/main/java/no/ssb/vtl/script/operations/filter/FilterOperation.java

```diff
@@ -46,21 +56,35 @@ public FilterOperation(Dataset dataset, VTLExpression predicate, ComponentBindin
         this.predicate = checkNotNull(predicate);
         this.componentBindings = checkNotNull(componentBindings);
     }
-    
+
     protected DataStructure computeDataStructure() {
         return getChild().getDataStructure();
     }
 
     @Override
-    public Stream<DataPoint> getData() {
+    public Stream<DataPoint> computeData(Ordering ordering, Filtering filtering, Set<String> components) {
         DataPointBindings dataPointBindings = new DataPointBindings(componentBindings, getDataStructure());
-        return getChild().getData()
-                .map(dataPointBindings::setDataPoint)
+
+        VtlOrdering childrenOrdering = (VtlOrdering) computeRequiredOrdering(ordering);
+        VtlFiltering childrenFiltering = (VtlFiltering) computeRequiredFiltering(filtering);
+
+        Stream<DataPoint> data = getChild().computeData(childrenOrdering, childrenFiltering, components);
```

Comment:
Assign the result from computeData to **original** directly instead of storing it temporarily in **data**.  The re-assignement of **data** is confusing.

Reply:
Refactoring gone wrong! Good catch.